src/parser/ParsedQuery.h

```diff
@@ -357,11 +359,29 @@ class ParsedQuery {
     [[nodiscard]] auto& getSelectVariables()  {
       return std::get<std::vector<string>>(_varsOrAsterisk);
     }
+
+    void pushVariablesOrder(const string& variable) {
+      if(!(std::find(_variablesOrder.begin(),
+                    _variablesOrder.end(),
+                    variable) != _variablesOrder.end())) {
+        _variablesOrder.emplace_back(variable);
+      }
+    }
+
+
+    [[nodiscard]] std::list<string>& retrieveOrder() {
+      return _variablesOrder;
+    }
```

Comment:
I think you should delete the non-const overload. If someone modifies the variables from outside,
this class is not valid anymore.